install-ci.yml

```diff
@@ -82,20 +82,88 @@
+        - libapache2-mod-wsgi
+      apache_mods_enabled:
+        - rewrite.load
       apache_vhosts:
         - name: logs
           server_name: logs.bonnyci.com
-          document_root: /var/www/bonny-logs
+          document_root: /var/www/bonny-logs/logs
           document_root_allow_override: None
           document_root_mode: "0755"
           document_root_options: "+Indexes +FollowSymLinks"
           vhost_extra: |
+            <Directory /etc/os_loganalyze/wsgi>
+              Require all granted
+            </Directory>
+
+            ErrorLog /var/log/apache2/os-loganalyze.log
+            CustomLog /var/log/apache2/os-loganalyze_access.log combined
+
+            RewriteEngine On
+
             AddType text/plain .log
             AddType text/plain .sh
             AddType text/plain .yml
             AddType text/plain .yaml
 
+            # use Apache to compress the results afterwards, to save on the wire
+            # it's approx 18x savings of wire traffic to compress. We need to
+            # compress by content types that htmlify can produce
+            AddOutputFilterByType DEFLATE text/plain text/html
+
+            <FilesMatch \.html\.gz$>
+              ForceType text/html
+              AddDefaultCharset UTF-8
+              AddEncoding x-gzip gz
+            </FilesMatch>
+
+            <FilesMatch \.css$>
+              # mod_mime_magic is sometimes passing css files as asm sources
+              # e.g css files generated by coverage reports
+              ForceType text/css
+            </FilesMatch>
+
+            # rewrite txt.gz & console.html[.gz] files to map to our internal htmlify
+            # wsgi app
+            # PT, Pass-through:  to come back around and get picked up by the
+            #                    WSGIScriptAlias
+            # NS, No-subrequest: on coming back through, mod-autoindex may have added
+            #                    index.html which would match the !-f condition. We
+            #                    therefore ensure the rewrite doesn't trigger by
+            #                    disallowing subrequests.
+            RewriteRule ^/(.*\.txt\.gz)$ /htmlify/$1 [QSA,L,PT,NS]
+            RewriteRule ^/(.*console\.html(\.gz)?)$ /htmlify/$1 [QSA,L,PT,NS]
+
+            # Check if the request exists as a file, directory or symbolic link
+            # If not, write the request to htmlify to see if we can fetch from swift
+            RewriteCond %{DOCUMENT_ROOT}%{REQUEST_FILENAME} !-f
+            RewriteCond %{DOCUMENT_ROOT}%{REQUEST_FILENAME} !-d
+            RewriteCond %{DOCUMENT_ROOT}%{REQUEST_FILENAME} !-l
+            RewriteCond %{REQUEST_FILENAME} !^/icon
+            RewriteRule ^/(.*)$ /htmlify/$1 [QSA,L,PT,NS]
+
+            WSGIDaemonProcess os-loganalyze processes=5 threads=1 display-name=%{GROUP}
+            WSGIScriptAlias /htmlify /etc/os_loganalyze/wsgi/os-loganalyze.py
+            WSGIApplicationGroup %{GLOBAL}
+
+            ServerSignature Off
+
```

Comment:
This seems like a LOT of data to be passing in at role call, a lot of configuration in the playbook itself rather than a inventory var or role var.